test/functional/wallet_create_tx.py

```diff
@@ -77,6 +79,51 @@ def test_tx_size_too_large(self):
         )
         self.nodes[0].settxfee(0)
 
+    def test_setfeerate(self):
+        self.log.info("Test setfeerate")
+        self.restart_node(0, extra_args=["-mintxfee=0.00003141"])  # 3.141 sat/vB
+        node = self.nodes[0]
+
+        def test_response(*, requested=0, expected=0, error=None, msg):
+            assert_equal(node.setfeerate(requested), {"wallet_name": "test_wallet", "fee_rate": expected, ("error" if error else "result"): msg})
+
+        # Test setfeerate with 10.0001 (CFeeRate rounding), "10.001" and "4" sat/vB
```

Comment:
Again I believe those check would be better places in a separate `setfeerate` test